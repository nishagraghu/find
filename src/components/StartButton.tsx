import React from 'react';
import { View, StyleSheet, Text, TouchableOpacity } from 'react-native';
import { getStarted } from '../utils/constants';
import { useNavigation } from '@react-navigation/native';
const StartButton : React.FC  = () => {
  const navigation = useNavigation <any>();

  const handleStartPress = () => {
    navigation.navigate('Mobil')
    
  };

  return (
    <View style={styles.container}>
      
      <TouchableOpacity testID="start-button" style={styles.button} onPress={handleStartPress}>
        
        <Text style={styles.buttonText}>{getStarted} </Text>
      </TouchableOpacity>
    </View>
  ); 
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  button: {
    marginTop: 20,
    height: 50,
    width: 300,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#1778F2',
    shadowColor: 'rgba(0,0,0,0.4)',
    shadowOffset: {
      width: 1,
      height: 5,
    },
    
    shadowOpacity: 0.34,
    shadowRadius: 6.27,
    elevation: 10,
  },
  buttonText: {
    width:200,

    color: 'white',
    fontWeight: 'bold',
    textAlign: 'center',
    
  },
});

export default StartButton;
