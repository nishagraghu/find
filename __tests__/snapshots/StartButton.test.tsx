import React from 'react';
import { render } from '@testing-library/react-native';

import StartButton from '../../src/components/StartButton'



jest.mock('@react-navigation/native', () => {
    const actualNav = jest.requireActual('@react-navigation/native');
    return {
      ...actualNav,
      useNavigation: jest.fn(),
    };
  });
describe('StartButton component', () => {
  test('renders correctly', () => {
    const { toJSON } = render(<StartButton />);
    expect(toJSON()).toMatchSnapshot();
  });
});
