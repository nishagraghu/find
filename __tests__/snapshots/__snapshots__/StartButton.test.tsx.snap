// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StartButton component renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#1778F2",
        "elevation": 10,
        "height": 50,
        "justifyContent": "center",
        "marginTop": 20,
        "opacity": 1,
        "shadowColor": "rgba(0,0,0,0.4)",
        "shadowOffset": {
          "height": 5,
          "width": 1,
        },
        "shadowOpacity": 0.34,
        "shadowRadius": 6.27,
        "width": 300,
      }
    }
    testID="start-button"
  >
    <Text
      style={
        {
          "color": "white",
          "fontWeight": "bold",
          "textAlign": "center",
          "width": 200,
        }
      }
    >
      LETS GET STARTED
       
    </Text>
  </View>
</View>
`;
